packages/lib/src/ref/core.ts

```diff
@@ -56,6 +62,28 @@ export function internalCustomRef<T extends object>(
       return this.resolver(this)
     }
 
+    onAttachedToRootStore() {
+      allRefs.add(this)
+
+      return () => {
+        allRefs.delete(this)
+      }
+    }
+
+    #savedOldTarget: T | undefined
```

Comment:
Just wanted to start using it, since they are transpiled to something really private instead of only private if you are using typescript. Do you see any cons?

Reply:
No, just wondering. Is there any performance overhead?

Reply:
Very minimal I think (when transpiled)